src/radical/pilot/pmgr/launching/default.py

```diff
@@ -1119,6 +1121,23 @@ def _prepare_pilot(self, resource, rcfg, pilot, expand):
             if virtenv_mode != 'private':
                 cleanup = cleanup.replace('v', '')
 
+        # don't stage sdists on 'installed', 'local'
+        if virtenv_mode in ['installed', 'local']: stage_sdists = False
+        else                                     : stage_sdists = True
+
+        # use local VE ?
+        if virtenv_mode == 'local':
+            if 'VIRTUAL_ENV' in os.environ:
```

Comment:
should we check that actual value is not empty? `if os.environ.get('VIRTUAL_ENV')`
no sure that it is really necessary, but kinda worry more for `'CONDA_PREFIX'`

Reply:
Ack, done.